### ISSUE TITLE
Tiles are missing when applying zappend to zarr files stored in s3

### ISSUE THREAD

**Issue by konstntokas** [bcdev/zappend#104]
**Describe the bug**
Zappend has been applied to append multiple zarr files along the time axis. The resulting cube shows random blocks being filled with nan values. Running it multiple times, the missing blocks are changing.
![Screenshot from 2024-10-28 09-12-43](https://github.com/user-attachments/assets/623a4485-71e5-412d-bf29-1bbeabcd9242)

**To Reproduce**
A data ID looks like `cubes/aux/era5_small/2016_11.zarr`. 
![Screenshot from 2024-10-28 09-15-25](https://github.com/user-attachments/assets/32aa4511-9340-479d-9aec-e49356911309)

The following zappend config was used. 

```python
config = {
    "target_dir": f"s3://{os.environ['S3_USER_STORAGE_BUCKET']}/{data_id_era5}",
    "target_storage_options": {
        "key": os.environ["S3_USER_STORAGE_KEY"],
        "secret": os.environ["S3_USER_STORAGE_SECRET"],
    },
    "slice_storage_options": {
        "key": os.environ["S3_USER_STORAGE_KEY"],
        "secret": os.environ["S3_USER_STORAGE_SECRET"],
    },
    "force_new": True,
    "disable_rollback": True,
    "logging": "DEBUG",
}



zappend(
    (
        open_zarr(f"s3://{os.environ['S3_USER_STORAGE_BUCKET']}/{data_id}")
        for data_id in data_ids[3]
    ),
    config=config,
)
```

**Expected behavior**
Zappend should just append the data cubes along a specific dimension without errors. 

**Python Environment**
 - operating system: DeepESDL (user environment built with conda store in DeepESDL)
 - zappend version, output of `zappend --version`: 0.8.0

**Comment by forman** [comment]
I don't think that this is actually a bug in zappend code, as it does not deal with individual chunks on its own, this is left to the preprocessing chain and the potential use of zarr / dask.

**Comment by forman** [comment]
Closing this. @konstntokas, please reopen if issue can be traced back into zappend or zappend can help avoiding this kind of issue.
